docs/plugins/development/creating-text-analysis-plugins.asciidoc

```diff
@@ -0,0 +1,130 @@
+[[creating-text-analysis-plugins]]
+=== Creating text analysis plugins
+
+Text analysis plugins provide {es} with custom {ref}/analysis.html[Lucene 
+analyzers, token filters, character filters, and tokenizers]. 
+
+[discrete]
+==== Plugin file structure
+
+Plugins are ZIP files composed of JAR files and two metadata files:
+
+* `stable-plugin-descriptor.properties` - a Java properties file that describes 
+the plugin. Refer to <<plugin-descriptor-file-{plugin-type}>>.
+* `named_components.json` - a JSON file mapping interfaces to key-value pairs 
+of component names and implementation classes.
+
+Note that only JAR files at the root of the plugin are added to the classpath
+for the plugin. If you need other resources, package them into a resources JAR.
+
+[discrete]
+==== The analysis plugin API
+
+Text analysis plugins can be written against the analysis plugin API. This API
+consists of the following dependencies:
+
+* `plugin-api`- an API used by plugin developers to implement custom {es} 
+plugins (stable).
```

Comment:
Everything here is stable, so I don't think the `(stable)` parenthetical is needed?